src/compile/model.ts

```diff
@@ -233,7 +233,11 @@ export abstract class Model {
   }
 
   public renameData(oldName: string, newName: string) {
-     this._dataNameMap.rename(oldName, newName);
+    if (oldName === newName) {
```

Comment:
why would this case happen?      Can we write code in a way that this does not happen?